src/Extensions/Microsoft.Diagnostics.Monitoring.Extension.Common/EgressHelper.cs

```diff
@@ -113,9 +128,39 @@ private static async Task GetStream(Stream outputStream, CancellationToken cance
         {
             const int DefaultBufferSize = 0x10000;
 
-            StdInStream = Console.OpenStandardInput();
             await StdInStream.CopyToAsync(outputStream, DefaultBufferSize, cancellationToken);
         }
+
+        // TODO: For .NET 7 and above, can use Stream.ReadExactly
+        private static async Task<int> ReadHelper(Memory<byte> buffer, long bufferSize, CancellationToken token)
+        {
+            Debug.Assert(bufferSize <= buffer.Length);
+
+            int totalRead = 0;
+            while (totalRead < bufferSize)
+            {
+                int read = await StdInStream.ReadAsync(buffer.Slice(totalRead), token).ConfigureAwait(false);
+                if (read == 0)
+                {
+                    break;
+                }
+
+                totalRead += read;
+            }
+
+            return totalRead;
+        }
+
+        private static async Task<byte[]> ReadFromStream(long bufferSize, CancellationToken token)
+        {
+            Memory<byte> buffer = new(new byte[bufferSize]);
+            if (bufferSize != await ReadHelper(buffer, bufferSize, token))
+            {
+                throw new EgressException(GenericEgressFailureMessage);
+            }
+
+            return buffer.ToArray();
```

Comment:
My understanding is that this makes a copy...is there a more efficient way I should be doing this, or for the amount of data we're working with is this not a concern?